src/Profile.hpp

```diff
@@ -167,9 +167,30 @@ namespace geopm
             /// application.
             virtual void epoch(void) = 0;
             virtual void shutdown(void) = 0;
-            virtual std::shared_ptr<ProfileThreadTable> tprof_table(void) = 0;
+            /// @brief Update the total work for one CPU and reset the
+            ///        work completed.  This method should be called
+            ///        by all threads in the same parallel region with
+            ///        the total expected to be completed by the
+            ///        entire group.
+            /// @param [in] cpu The Linux logical CPU obtained with
+            ///             get_cpu().
+            /// @param [in] num_work_unit The total work units for all
+            ///             threads in the same parallel region.
+            virtual void thread_init(int cpu, uint32_t num_work_unit) = 0;
```

Comment:
Was the change in precision (from how it used to be with size_t's in init_prof_loop, to all uint32_t's) just made to align everything to uint32_t?

Is this enough precision?

Reply:
This is to enable a lock free implementation, we need to stick to 32 bit integers which are atomic.  Note the hints are shifted right 32 bits so they fit.